core/src/main/java/com/redhat/lightblue/client/Expression.java

```diff
@@ -0,0 +1,52 @@
+package com.redhat.lightblue.client;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ContainerNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+/**
+ * Base class for all expressions. Contains a container json node,
+ * either an object, or an array
+ */
+public abstract class Expression extends ExpressionPart {
+
+    /**
+     * Construct expression with the given object or array node
+     */
+    protected Expression(ContainerNode node) {
+        super(node);
+    }
+
+    /**
+     * Construct an expression with an empty array or object node
+     */
+    protected Expression(boolean arrayNode) {
+        super(arrayNode?JsonNodeFactory.instance.arrayNode():
+              JsonNodeFactory.instance.objectNode());
+    }
+
+    /**
+     * Add field:value
+     */
+    public Expression add(String field,JsonNode value) {
+        ((ObjectNode)node).set(field,value);
```

Comment:
Rewrote the exception as runtime exception.

On Thu, Aug 6, 2015 at 12:08 PM, Dennis Crissman <notifications@github.com>
wrote:

> In core/src/main/java/com/redhat/lightblue/client/Expression.java
> <https://github.com/lightblue-platform/lightblue-client/pull/144#discussion_r36445919>
> :
>
> > +        super(node);
> > +    }
> > +
> > +    /**
> > +     * Construct an expression with an empty array or object node
> > +     */
> > +    protected Expression(boolean arrayNode) {
> > +        super(arrayNode?JsonNodeFactory.instance.arrayNode():
> > +              JsonNodeFactory.instance.objectNode());
> > +    }
> > +
> > +    /**
> > +     * Add field:value
> > +     */
> > +    public Expression add(String field,JsonNode value) {
> > +        ((ObjectNode)node).set(field,value);
>
> I would say do an instanceof check and throw a meaningful exception.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-client/pull/144/files#r36445919>
> .
>